File: golang/xrsa/crypt_test.go
package xrsa

import(
	"testing"
)

func TestEd(t *testing.T) {
	a := "Hello"
	key := "123456"

	b := ed([]byte(a), key)
	c := ed(b, key)
	if string(a) != string(c) {
		t.Fatal("failed")
	}
}

func TestEncryptDecryptt(t *testing.T) {
	a := "Hello"
	key := "123456"

	crypt := NewCrypt(key)
	encrypted := crypt.Encrypt([]byte(a))

	decrypted := crypt.Decrypt(encrypted)
	if string(decrypted) != a {
		t.Fatal("encrypt decrypt failed")
	}
}

File: php/tests/CryptTest.php
<?php

use XRsa\Crypt;
use PHPUnit\Framework\TestCase;

class CryptTest extends TestCase
{
    public function testEncryptDecrypt()
    {
        $crypt = new Crypt(123456);
        $data = "Hello, World";
        $encrypted = $crypt->encrypt($data);
        $decrypted = $crypt->decrypt($encrypted);

        $this->assertEquals($data, $decrypted);
    }
}
